spark/v3.4/spark/src/main/java/org/apache/iceberg/spark/source/BaseReader.java

```diff
@@ -279,5 +284,29 @@ protected void markRowDeleted(InternalRow row) {
         counter().increment();
       }
     }
+
+    @Override
+    protected DeleteLoader newDeleteLoader() {
+      return new CachingDeleteLoader(this::loadInputFile);
+    }
+
+    private class CachingDeleteLoader extends BaseDeleteLoader {
+      private final SparkExecutorCache cache;
+
+      CachingDeleteLoader(Function<DeleteFile, InputFile> loadInputFile) {
+        super(loadInputFile);
+        this.cache = SparkExecutorCache.getOrCreate();
+      }
+
+      @Override
+      protected boolean canCache(long size) {
+        return cache != null && size < cache.maxEntrySize();
```

Comment:
Nit: It's a bit pedantic considering what it's for, but just realized shouldn't this be `size <= cache.maxEntrySize()`` instead of just < ? Based on the current description of the cache property 

```
  /** Returns the max entry size in bytes that will be considered for caching. */
```

Reply:
Sounds good, I'll update this in both places in a follow-up PR to keep this one as a cherry-pick.